### PR TITLE
Add prometheus and grafana to the Helm chart

### PR DESCRIPTION
**Pull request by metlos**
### What does this PR do?

This PR adds prometheus and grafana to the Che's Helm chart.

The deployment of the new components is conditional on the `global.metricsEnabled` chart property which is also used for the for enabling the metrics in Che server itself (by passing the value in the `CHE_METRICS_ENABLED` env var).

Prometheus is preconfigured to scrape the Che's metrics endpoint and grafana is preconfigured with a prometheus datasource reading Che's metrics and an example dashboard reading the Che server's heap characteristics.

The PR also moves Che's custom charts into "custom-charts" directory so that they don't interfere with helm which uses the "charts" directory to download and use all the other remote dependencies.

### What issues does this PR fix or reference?

#12059 - Deploy Prometheus in Che Helm Chart

#### Docs PR
* docs will be done later
* this will be in the release notes for Che 6.16

### CODE REVIEW

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Review by garagatyi — COMMENTED** [review]
Looks good. What is the reason why this PR is in WIP?

**Comment by metlos** [comment]
> Looks good. What is the reason why this PR is in WIP?
I wanted to make sure I have the ingress issue sufficiently solved. It's WIP no more now..

**Review by garagatyi — APPROVED** [review]

**Comment by skabashnyuk** [comment]
ci-build

**Comment by skabashnyuk** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1269/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1269//Selenium_20tests_20report/)
docker image: eclipseche/che-server:12088
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by skabashnyuk** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1271/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1271//Selenium_20tests_20report/)
docker image: eclipseche/che-server:12088
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by skabashnyuk — APPROVED** [review]

**Review by sleshchenko — APPROVED** [review]
LGTM

**Comment by sleshchenko** [comment]
One more comment: PR description has TODO for Docs PR. Please make you did it/remove if it is not needed anymore.

File: deploy/kubernetes/helm/che/templates/configmap.yaml [hunk 1]
```diff
@@ -97,3 +97,5 @@ data:
   JAEGER_SAMPLER_TYPE: "const"
   JAEGER_SAMPLER_PARAM: "1"
   JAEGER_REPORTER_MAX_QUEUE_SIZE: "10000"
+
```
[the remark this answers is not in the document]
Comment (metlos) [reply deploy/kubernetes/helm/che/templates/configmap.yaml]: I personally didn't even notice it. I'll remove it.

**Comment by skabashnyuk** [comment]
ci-build

**Pull request merged**